### DISCUSSION TITLE
Code first migration EF Core

### DISCUSSION THREAD

**Discussion by milovidov983** [thepirat000/Audit.NET#570] [Q&A]
Hello everyone. I use the EF Core library and I need the audit some tables.

Correct me if I'm wrong:
 
In order for everything to work, I need to implement a DbSet with this entity(https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/EventEntry.cs) in my context and call it Audit?

Something like this?

```csharp
public class CustomerDbContext : AuditDbContext
{
    public virtual DbSet<Entities.MyEntities > MyEntities { get; set; }
    public virtual DbSet<Audit.EntityFramework.EventEntry> Audit { get; set; } <------ like this ?
    ...
}
```

**Comment by thepirat000** [comment]
It's not like that. EventEntry is not intended to be used as an entity class in a DbSet. Neither the library will detect the name of your DbSet. You must configure the audit output (Data Provider).

To configure the output of your audits, you can use any of the provided [Data Providers](https://github.com/thepirat000/Audit.NET/blob/master/README.md#data-providers-included), which in your case it seems that you will use the [EntityFramework DataProvider](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md#entity-framework-data-provider) to store the audit events in the same DbContext you're auditing. 

So you have to define the structure of the audit table(s) since the library will make no assumptions at all. 

You could want to store the entire event in a single JSON column, or maybe have one audit table for each audited entity, maybe you just want some of the properties of the audit event, etc... You can see examples for the different configurations [here](https://github.com/thepirat000/Audit.NET/blob/master/src/Audit.EntityFramework/README.md#entity-framework-data-provider),

**Comment by milovidov983** [comment]
thx
